vars/buildImage.groovy

```diff
@@ -0,0 +1,99 @@
+// Create a Job and BuildConfig Template to trigger the Build
+// Available parameters:
+//    dockerFile    string -- DockerFile used for the buildconfig
+//    gitURL:       string -- Git URL used for the buildconfig
+//    image         string -- Image used for the buildconfig
+//    nameSpace:    string -- Namespace name to run the build
+def call(params = [:]) {
+    def imageName
+    pipeline {
+        // We don't need any specify agent just to add the buildconfig and trigger it,
+        // this pod will be destroyed in the end
+        agent {
+            node { label 'nodejs' }
+        }
+        stages {
+            stage('Prepare Env') {
+                steps{
+                   script{
+                       openshift.withCluster() {
+                           openshift.withProject() {
+
+                               def bcJSON  = libraryResource 'com/github/coreos/buildconfig.json'
+                               def jobJSON = libraryResource 'com/github/coreos/job.json'
+                               def jobObj  = readJSON text: jobJSON
+                               def bcObj   = readJSON text: bcJSON
+                               def tagName = UUID.randomUUID()
+
+                               jobObj['parameters'][0] +=['name': 'NAME', 'value': "cleanup-${tagName}".toString()]
+
+                               writeYaml(file: "job-${tagName}.yaml", data: jobObj, overwrite: true)
+                               jobTemplatePath = env.WORKSPACE+"/job-${tagName}.yaml"
+
+                               def app = openshift.newApp(jobTemplatePath)
+                               def job = app.narrow('job')
+                               def jobMap = job.object()
+                               def uid =  jobMap.metadata.get('uid')
+
+                               //Create and unique name for image and bc
+                               imageName = "bc-${tagName}"
+                               bcObj['parameters'][0] +=['name': 'NAME', 'value': "bc-${tagName}".toString()]
+
+                               // Add OwnerReference for cascading deletion with the Job timeout
+                               bcObj['parameters'][8] +=['name': 'UID', 'value': "${uid}".toString()]
+                               bcObj['parameters'][7] +=['name': 'OWNERNAME', 'value': "cleanup-${tagName}".toString()]
+
+                               if (params['namespace']) {
+                                   bcObj['parameters'][1] +=['name': 'NAMESPACE', 'value': "${params['nameSpace']}"]
+                               }
+                               if (params['gitURL']) {
+                                   bcObj['parameters'][2] +=['name': 'SOURCE_REPOSITORY_URL', 'value': "${params['gitURL']}"]
+                               }
+                               if (env.BRANCH_NAME && env.BRANCH_NAME.startsWith('PR-')) {
```

Comment:
Oh, does this work? I've had trouble in the past accessing environment variables from Jenkins shared libraries.

Reply:
I didn't test using a PR. However I'm able to get env.BRANCH_NAME from the branch I'm triggering it. So I get `origin/lib` in my case. Looking at the documentation when it is triggered from a PR the format will be PR-number